### ISSUE TITLE
skipEmptyPeriods does not return correct dataItem indexes 

### ISSUE THREAD

**Issue by jimmykane** [amcharts/amcharts4#1827]
I use an adapter on 'dy' for a Date axis to offset the labels

```
 categoryAxis.renderer.labels.template.adapter.add('dy', (dy, target) => {
      if (this.vertical && target.dataItem && target.dataItem.index % 2) {
        return dy + 20;
      }
      return dy;
    });
```
This plays fine as seen below 

<img width="893" alt="Screenshot 2019-11-01 at 12 09 07" src="https://user-images.githubusercontent.com/1268723/68021065-b8fbfe00-fca0-11e9-92e3-89a963af691a.png">

When using `skipEmptyPeriods` on a Date axis this happens. 

<img width="904" alt="Screenshot 2019-11-01 at 12 12 36" src="https://user-images.githubusercontent.com/1268723/68021151-ecd72380-fca0-11e9-9cdb-535942a505e5.png">

Apparently the index is not "updated" to the new array? 

Is this a bug? Can it be a workarround?

**Comment by zeroin** [comment]
`index` is totally unreliable in this case, as it only shows when the data item was created, so it might change while zooming etc. However I understand knowing actual index might be useful, like in your case, so I added `itemIndex` to `AxisDataItem` and it will soon be available. I also noticed that toggling of `skipEmptyPeriods` was not working, which, I guess could be a problem in your case, so this will also be fixed in next release.

**Comment by martynasma** [comment]
Fixed in [4.7.8](https://github.com/amcharts/amcharts4/releases/tag/4.7.8).

## [4.7.8] - 2019-11-05

### Added
- `itemIndex` added to `AxisDataItem`. Shows current index of data item on the axis.

### Changed
- `extremeschanged` is now dispatched by `DateAxis` when the `min`/`max` are animating.

### Fixed
- Setting `maskBullets` on an `XYChart` will toggle on or off bullet masking dynamically.
- In some particular cases some Sankey nodes were drawn out of bounds.
- `behaviorcanceled` event of `ChartCursor` was not being fired.
- Regression plugin was failing when added to Series object that was not yet attached to a Chart.
- JSON config was failing where Label's `text` or `html` properties were being set to percent value.
- Toggling of `DateAxis.skipEmptyPeriods` was not working after the chart was already inited.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
